core/coreapi/unixfs.go

```diff
@@ -94,7 +94,9 @@ func (api *UnixfsAPI) Add(ctx context.Context, files files.Node, opts ...options
 	fileAdder.RawLeaves = settings.RawLeaves
 	fileAdder.NoCopy = settings.NoCopy
 	fileAdder.Name = settings.StdinName
+	fileAdder.BaseName = settings.BaseName
 	fileAdder.CidBuilder = prefix
+	fileAdder.TopHidden = settings.TopHidden
```

Comment:
See https://github.com/ipfs/interface-go-ipfs-core/issues/17. I'm not sure if we really need this.

(yes, it'll be a breaking change for people using the HTTP API manually but I'm pretty sure they'd be _very_ surprised at the current behavior).